### DISCUSSION TITLE
Is it possible use dnscrypt-proxy in Android device?

### DISCUSSION THREAD

**Discussion by taicaile** [DNSCrypt/dnscrypt-proxy#1784] [Q&A]
Hello, I am wondering if it is possible to run dnscrypt-proxy in Android device? and is there tutorial how I can set up it?

Currently I am using my Android share wifi via hotspot, and I want to set a dns server in Android to enable DOH globally.

**Comment by jedisct1** [comment]
There's an entire thread on this: https://github.com/DNSCrypt/dnscrypt-proxy/discussions/1785

**Reply by taicaile** [reply]
> There's an entire thread on this: #1785

Hi, thanks. I just read that thread and I couldn't ask question there since it has been closed.

I have tired some android app(they make a virtual vpn) in my Android 8 device, and it works for browsering activities in Android device locally. when I share the cellular data as hotspot, and other user connect it, the dns setting is not working. I don't why.

**Reply by d3cim** [reply]
Hi, I've been mantaining `dnscrypt-proxy` on Android for two years now .. since [bluemeda](https://github.com/bluemeda/dnscrypt-proxy-magisk) stopped developing his module.

The only requirement is that you have `root` permission on your device. For more information, also regarding the installation process, please check my repository:

https://codeberg.org/quindecim/dnscrypt-proxy-android

I'm not active on GitHub at all, for this reason I take the opportunity to thank @jedisct1 for the project, all the contributors and all the server mantainers.

Keep it up!

**Comment by d3cim** [comment]
Moved to GitHub https://github.com/quindecim/dnscrypt-proxy-android
